core/services/pipeline/graph.go

```diff
@@ -65,6 +65,11 @@ func (g TaskDAG) TasksInDependencyOrder() ([]Task, error) {
 			return nil, err
 		}
 
+		err = task.ApplyDefaults(node.attrs, g, *node)
```

Comment:
nit: could move this above the UnmarshalTaskFromMap avoid the "set default if not set" and just have it naturally override the default assuming that mapstructure overrides